### DISCUSSION TITLE
Usage with AWS ECS Fargate: Connection pool-size for VCPU's?

### DISCUSSION THREAD

**Discussion by JMitnik** [prisma/prisma#9273] [Q&A] [answered]
Hi folks,

I assume I might now be the only one who uses prisma with aws ecs containers. While it works, I realize the connection is pretty much _chugging_, and I wonder whether this is because of some setting in the containers. When reading about connection pooling, I see that the recommended amount of connections is _numPhysicalCores * 2 + 1_. 

However, in ECS Fargate (which is essentially serverless), there is no mention of physical cores, but VCPU's. Therefore, I find it a bit ambiguous whether the number of VCPU's is in any way indicative of the size of connection pool.

So my question is, does anyone know how this maps to physical cores?

**Comment by ryands17** [comment, marked as answer]
@JMitnik 👋 

VCPU's would be equivalent to CPU so that connection limit would be set the same as above. You can also optimise the settings as shown [here](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#optimizing-the-connection-pool) if you're facing any issues.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by alan345** [comment]
Hey @JMitnik 


I do actually have the same issue. 

`Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 3)
`
I know it was a long time ago, but do you remember how did you solve this? Did you override the url connection with a new connection pool? How much was your value for the connection pool?

I don't know how many cpu my fargate instance has, (as it is serverless)

I am using fargate and MySQL 

Thanks for your help.

**Reply by JMitnik** [reply]
Hi @alan345!

I'm afraid we shortly after moved away from prisma (not for this reason), so I don't have a 100% confident answer for you. 

But I do recall that when Fargate killed a container, it would send a signal (IIRC it was SIGTERM or SIGKILL). We attached a listener to that event in our `node` server, and then fetched our Prisma singleton and manually disabled it. That seemed to work as patch, but I never was satisfied with that result because it wasn't "clean".

I'd advise you to check if manually `$disconnect` fixes the problem somewhat.

**Reply by alan345** [reply]
Thanks! Super useful 

Which solution are you using now?

**Reply by JMitnik** [reply]
Microservice restructure, eventually we adapted the Laravel backend :)
